src/request_handler.js

```diff
@@ -208,6 +208,10 @@ export function updateAccessToken () {
       this.tokenExpiration = Date.now() + (tokenInfo.expires_in * 1000);
       if (tokenInfo.error === 'invalid_grant') {
         throw new Error('"Invalid grant" error returned from reddit. (You might have incorrect credentials.)');
+      } else if (tokenInfo.error_description !== undefined) {
+        throw new Error(`Reddit return an error. ${tokenInfo.error}: ${tokenInfo.error_description}`);
+      } else if (tokenInfo.error !== undefined) {
+        throw new Error(`Reddit return an error: ${tokenInfo.error}`);
```

Comment:
Nitpick: I think these should say "Reddit returned an error" (with "ed" at the end of "return").